src/Sylius/Bundle/ThemeBundle/Factory/ThemeFactory.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Sylius\Bundle\ThemeBundle\Factory;
+
+use Sylius\Bundle\ThemeBundle\Model\ThemeInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+class ThemeFactory implements ThemeFactoryInterface
+{
+    /**
+     * @var string
+     */
+    private $themeClassName;
+
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    /**
+     * @param string $themeClassName
+     * @param PropertyAccessorInterface $propertyAccessor
+     */
+    public function __construct($themeClassName, PropertyAccessorInterface $propertyAccessor)
+    {
+        $this->themeClassName = $themeClassName;
+        $this->propertyAccessor = $propertyAccessor;
+
+        $this->optionsResolver = new OptionsResolver();
+        $this->optionsResolver
+            ->setRequired([
+                'name',
+                'logical_name',
+            ])
+            ->setDefined([
+                'description',
+            ])
+            ->setDefaults([
+                'parents' => [],
+            ])
+            ->setAllowedTypes('parents', 'array')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createFromArray(array $themeData)
+    {
+        /** @var ThemeInterface $theme */
+        $theme = new $this->themeClassName();
+
+        $themeData = $this->optionsResolver->resolve($themeData);
+
+        foreach ($themeData as $attributeKey => $attributeValue)
+        {
+            $attributeKey = $this->normalizeAttributeKey($attributeKey);
+
+            try {
+                $this->propertyAccessor->setValue($theme, $attributeKey, $attributeValue);
+            } catch (NoSuchPropertyException $exception) {
+                // Ignore properties that does not exist in given theme model.
```

Comment:
I'm not sure about ignoring exceptions. I understand what you wanted to achieve here, but ignoring and commenting exceptions is not the best solution. It happened not only here.